File: 204.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 3:
            return 0
        prime_set = [1] * n
        prime_set[0] = 0
        prime_set[1] = 0
        for i in range(2,int(n**0.5) + 1):
            if prime_set[i]:
                prime_set[i*2:n:i] = [0]*len(prime_set[i*2:n:i])
        return sum(prime_set)
print(Solution().countPrimes(499979))

File: 204.java
class Solution {
    public int countPrimes(int n) {
        if (n <= 2) return 0;
        int res = 0;
        for (int i = 2; i < n; i++) {
            if (isPrime(i)) res++;
        }
        return res;
    }

    private boolean isPrime(int n) {
        int m = (int) Math.sqrt(n);
        for (int i = 2; i <= m; i++) {
            if (n % i == 0 && n != i) return false;
        }
        return true;
    }
}